Documentation/teaching/so2/assign0-kernel-api.rst

```diff
@@ -0,0 +1,120 @@
+=========================
+Assignment 0 - Kernel API
+=========================
+
+-  Deadline: :command:`Wednesday, 24 March 2020, 23:59`
+
+Assignment's Objectives
+=======================
+
+*  getting familiar with the qemu setup
+*  inserting / deleting kernel modules
+*  getting familiar with the list API implemented in the kernel
+*  have fun :)
+
+Statement
+==============
+
+Write a kernel module called `list` (the resulting file must be called `list.ko`) which stores data (strings)
+in an internal list.
+
+It is mandatory to use `the list API <https://github.com/torvalds/linux/blob/master/include/linux/list.h>`__
+implemented in the kernel.
+For details you can take a look `the laboratory 3 <https://ocw.cs.pub.ro/courses/so2/laboratoare/lab03#liste>`__
+
+It is mandatory to start the implementation of the assignment from the code skeleton found in 
+`list.c <https://github.com/linux-kernel-labs/linux/blob/master/tools/labs/templates/assignments/0-list/list.c>`__ file.
+
+Use the new test infrastructure by running the commands below:
+
+.. code-block:: console
+
+    $ git clone https://github.com/linux-kernel-labs/linux.git
+    $ cd linux/tools/labs
+    $ LABS=assignments/0-list make skels
+    $ # The code skeleton will be found in api_assignment/0-list/list.c
+    $ make build
+    $ make copy
+    $ make boot
+
+The module exports a directory named :command:`list` to procfs. The directory contains two files:
+
+-   :command:`management`: with write-only access; is the interface for transmitting commands to the kernel module
+-   :command:`preview`: cu acces read-only; with read-only access; is the interface through which the internal contents of the kernel list can be viewed.
+
+`The code skeleton <https://github.com/linux-kernel-labs/linux/blob/master/tools/labs/templates/assignments/0-list/list.c>`__ implements the two procfs files.
+You will need to create a list and add support for `adding` and `reading` data. Follow the TODOs in the code for details. 
```

Comment:
To avoid repetition use "create a list and implement support for adding"

Reply:
Fixed.